integration/test-integration/src/test/java/org/wso2am/micro/gw/tests/apim/APIMLifecycleBaseTest.java

```diff
@@ -0,0 +1,771 @@
+        assertEquals(apiRevisionsFullGetResponse.getResponseCode(), HTTP_RESPONSE_CODE_OK,
+                     "Unable to retrieve revisions" + apiRevisionsFullGetResponse.getData());
+        List<JSONObject> revisionFullList = new ArrayList<>();
+        JSONObject jsonFullObject = new JSONObject(apiRevisionsFullGetResponse.getData());
+
+        JSONArray arrayFullList = jsonFullObject.getJSONArray("list");
+        for (int i = 0, l = arrayFullList.length(); i < l; i++) {
+            revisionFullList.add(arrayFullList.getJSONObject(i));
+        }
+        for (JSONObject revision : revisionFullList) {
+            revisionUUID = revision.getString("id");
+            HttpResponse apiRevisionsDeleteResponse = restAPIPublisher.deleteAPIProductRevision(apiId, revisionUUID);
+            assertEquals(apiRevisionsDeleteResponse.getResponseCode(), HTTP_RESPONSE_CODE_OK,
+                         "Unable to delete API Product Revisions:" + apiRevisionsDeleteResponse.getData());
+        }
+
+        //Waiting for API un-deployment
+        waitForAPIDeployment();
+        return revisionUUID;
+    }
+
+    /**
+     * Return a String with combining the value of API Name, API Version, and API Provider Name as key:value format
+     *
+     * @param api - Instance of API object that includes the API Name, API Version, and API Provider Name to create the
+     *            String
+     * @return String - with API Name,API Version and API Provider Name as key:value format
+     */
+    private String getAPIIdentifierString(API api) {
+        return " API Name:" + api.getName() + ", API Version:" + api.getVersion() +
+                ", API Provider Name :" + api.getProvider() + " ";
+
+    }
+
+    /**
+     * Return a String with combining the value of API Name, API Version, and API Provider Name as key:value format.
+     *
+     * @param apiRequest - Instance of APIRequest object  that include the  API Name,API Version and API Provider Name
+     *                   to create the String
+     * @return String - with API Name, API Version, and API Provider Name as key:value format
+     */
+    protected String getAPIIdentifierStringFromAPIRequest(APIRequest apiRequest) {
+        return " API Name:" + apiRequest.getName() + " API Version:" + apiRequest.getVersion() +
+                " API Provider Name :" + apiRequest.getProvider() + " ";
+    }
+
+    /**
+     * Retrieve  the value from JSON object bu using the key.
+     *
+     * @param httpResponse - Response that containing the JSON object in it response data.
+     * @param key          - key of the JSON value the need to retrieve.
+     * @return String - The value of provided key as a String
+     * @throws MicroGWTestException - Exception throws when resolving the JSON object in the HTTP response
+     */
+    protected String getValueFromJSON(HttpResponse httpResponse, String key) throws MicroGWTestException {
```

Comment:
Shall we move methods similar to these to a suitable file inside the utils folder and make them static?

Reply:
+1